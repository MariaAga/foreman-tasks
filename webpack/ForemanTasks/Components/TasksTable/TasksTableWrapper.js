import React from 'react';
import { getURIQuery } from 'foremanReact/common/helpers';
import { ExportButton } from './Components/ExportButton';
import TasksTable from './TasksTable';
import { addSearchToURL } from './TasksTableHelpers';

const TasksTableWrapper = ({ ...props }) => (
  <div className="tasks-table-wrapper">
    <ExportButton
      url={addSearchToURL(
        '/foreman_tasks/tasks.csv',
        getURIQuery(window.location.href)
      )}
    />
    <TasksTable {...props} />
  </div>
);

export default TasksTableWrapper;
