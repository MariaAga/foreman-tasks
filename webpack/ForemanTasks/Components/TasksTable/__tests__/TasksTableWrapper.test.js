import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
import TasksTableWrapper from '../TasksTableWrapper';
import fixtures from './TasksTable.fixtures';

const firstFixture = {
  [Object.keys(fixtures)[0]]: fixtures[Object.keys(fixtures)[0]],
};

describe('TasksTableWrapper', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TasksTableWrapper, firstFixture));
});
