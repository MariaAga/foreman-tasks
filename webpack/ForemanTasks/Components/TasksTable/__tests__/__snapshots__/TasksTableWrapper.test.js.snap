// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksTableWrapper rendering render with minimal Props 1`] = `
<div
  className="tasks-table-wrapper"
>
  <ExportButton
    url="/foreman_tasks/tasks.csv?include_permissions=true"
  />
  <TasksTable
    error={null}
    getTableItems={[MockFunction]}
    itemCount={0}
    loading={false}
    pagination={
      Object {
        "page": 1,
        "perPage": 10,
      }
    }
    query={Object {}}
    results={
      Array [
        "a",
      ]
    }
    status="PENDING"
  />
</div>
`;
